src/main/java/org/apache/pirk/responder/wideskies/ResponderDriver.java

```diff
@@ -50,103 +40,31 @@
+    String platformName = SystemConfiguration.getProperty(ResponderProps.PLATFORM, "None");
+    logger.info("Attempting to use platform {} ...", platformName);
     try
     {
-      platform = Platform.valueOf(platformString.toUpperCase());
-    } catch (IllegalArgumentException e)
-    {
-      logger.error("platform " + platformString + " not found.");
-    }
-
-    logger.info("platform = " + platform);
-    switch (platform)
-    {
-      case MAPREDUCE:
-        logger.info("Launching MapReduce ResponderTool:");
-
-        ComputeResponseTool pirWLTool = new ComputeResponseTool();
-        ToolRunner.run(pirWLTool, new String[] {});
-        break;
-
-      case SPARK:
-        logger.info("Launching Spark ComputeResponse:");
-
-        ComputeResponse computeResponse = new ComputeResponse(FileSystem.get(new Configuration()));
-        computeResponse.performQuery();
-        break;
-
-      case SPARKSTREAMING:
-        logger.info("Launching Spark ComputeStreamingResponse:");
-
-        ComputeStreamingResponse computeSR = new ComputeStreamingResponse(FileSystem.get(new Configuration()));
-        try
-        {
-          computeSR.performQuery();
-        } catch (SystemExitException e)
-        {
-          // If System.exit(0) is not caught from Spark Streaming,
-          // the application will complete with a 'failed' status
-          logger.info("Exited with System.exit(0) from Spark Streaming");
-        }
-
-        // Teardown the context
-        computeSR.teardown();
-        break;
-
-      case STORM:
-        logger.info("Launching Storm PirkTopology:");
-        PirkTopology.runPirkTopology();
-        break;
-
-      case STANDALONE:
-        logger.info("Launching Standalone Responder:");
-
-        String queryInput = SystemConfiguration.getProperty("pir.queryInput");
-        Query query = new LocalFileSystemStore().recall(queryInput, Query.class);
-
-        Responder pirResponder = new Responder(query);
-        pirResponder.computeStandaloneResponse();
-        break;
-    }
-  }
-
-  // Exception and Security Manager classes used to catch System.exit from Spark Streaming
-  private static class SystemExitException extends SecurityException
-  {}
-
-  private static class SystemExitManager extends SecurityManager
-  {
-    @Override
-    public void checkPermission(Permission perm)
-    {}
-
-    @Override
-    public void checkExit(int status)
-    {
-      super.checkExit(status);
-      if (status == 0) // If we exited cleanly, throw SystemExitException
+      ResponderPlugin responder = ResponderService.getInstance().getResponder(platformName);
+      if (responder == null)
       {
-        throw new SystemExitException();
+        logger.error("No such platform plugin found: {}!", platformName);
       }
       else
       {
-        throw new SecurityException();
+        responder.run();
       }
-
+    }
+    catch (PIRException pirEx)
```

Comment:
IIRC the PIRException was being thrown with the original as a cause, so you won't loose the stacktrace.  I agree that tidying up exceptions is beyond this PR, and will require some 'waves' of wrapping Exception until they are banished.